File: src/lessons/welcome/MethodsEntity.java
package lessons.welcome;

public class MethodsEntity extends jlm.universe.bugglequest.SimpleBuggle {
	@Override
	public void forward(int i)  { 
		throw new RuntimeException("forward(int) forbidden in this exercise");
	}

	@Override
	public void backward(int i) {
		throw new RuntimeException("backward(int) forbidden in this exercise");
	}

	/* BEGIN SOLUTION */
	public void goAndGet() {
		int i = 0;
		while (!isOverBaggle()) {
			i++;
			forward();
		}
		pickUpBaggle();
		while (i>0) {
			backward();
			i--;
		}
		dropBaggle();
	}
	/* END TEMPLATE */

	@Override
	public void run() { 
		for (int i=0; i<7; i++) {
			goAndGet();
			turnRight();
			forward();
			turnLeft();
		}
	} 
}

File: src/lessons/welcome/MethodsEntity.py
def forward(i=-1):
    if i==-1:
      entity.forward()
    else:
      errorMsg("Sorry Dave, I cannot let you use forward with argument")
def backward(i=-1):
    if i==-1:
      entity.backward()
    else:
      errorMsg("Sorry Dave, I cannot let you use backward with argument")
for i in range(7):
  goAndGet()
  turnRight()
  forward()
  turnLeft()
# BEGIN SOLUTION
def goAndGet():
  cpt = 0
  
  while isOverBaggle() == False:
    cpt = cpt+1
    forward()
  pickUpBaggle()
  for i in range(cpt):
    backward()
  dropBaggle()
# END TEMPLATE 
